File: src/dotnet/HQ.Abstractions/Users/GetUsersV1.cs
using HQ.Abstractions.Common;

namespace HQ.Abstractions.Users;

public class GetUsersV1
{
    public class Request : PagedRequestV1
    {
        public string? Search { get; set; }
    }

    public class Response : PagedResponseV1<Record>;
    public class Record
    {
        public Guid Id { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public string Email { get; set; } = null!;
        public Guid? StaffId { get; set; }
        public bool Enabled { get; set; }
        public bool IsStaff { get; set; }
        public bool IsManager { get; set; }
        public bool IsPartner { get; set; }
        public bool IsExecutive { get; set; }
        public bool IsAdministrator { get; set; }
    }
}

File: src/angular/hq/src/app/models/users/get-users-v1.ts
import { PagedRequestV1 } from '../common/paged-request-v1';
import { PagedResponseV1 } from '../common/paged-response-v1';

export interface GetUsersRequestV1 extends PagedRequestV1 {
  search?: string | null;
  id?: string | null;
}

export interface GetUsersRecordV1 {
  id: string;
  firstName?: string | null;
  lastName?: string | null;
  email: string | null;
  staffId?: string | null;
  enabled: boolean;
  isStaff: boolean;
  isManager: boolean;
  isPartner: boolean;
  isExecutive: boolean;
  isAdministrator: boolean;
}

export interface GetUsersRecordsV1 {
  records: [GetUsersRecordV1];
  total: number;
}

export interface GetUsersResponseV1 extends PagedResponseV1<GetUsersRecordV1> {}
